File: nodes/standard-deviation/standard-deviation.process.js
export default async ({inputs, settings, config}) => {

  const numbers = Array.isArray(inputs.numbers) ? inputs.numbers : [inputs.numbers];
  
  // Validate all inputs are numbers
  for (const n of numbers) {
    if (n === null || n === undefined || isNaN(Number(n))) {
      throw new Error("Input contains non-numeric values that cannot be processed");
    }
  }
  
  // Convert all values to numbers
  const validNumbers = numbers.map(n => Number(n));
  
  if (validNumbers.length === 0) {
    return { 
      result: 0,
      variance: 0
    };
  }
  
  // Calculate mean
  const mean = validNumbers.reduce((sum, val) => sum + val, 0) / validNumbers.length;
  
  // Calculate variance
  const variance = validNumbers.reduce((sum, val) => {
    const diff = val - mean;
    return sum + (diff * diff);
  }, 0) / validNumbers.length;
  
  return {
    result: Math.sqrt(variance),
    variance: variance
  };
}; 

File: nodes/standard-deviation/standard-deviation.process.py
import math

async def process(inputs, settings, config, nodeConfig):
  numbers = inputs.get("numbers", [])
  if not isinstance(numbers, list):
    numbers = [numbers]
    
  # Validate all inputs are numbers
  for n in numbers:
    try:
      float(n)
    except (ValueError, TypeError):
      raise ValueError("Input contains non-numeric values that cannot be processed")
  
  # Convert all values to numbers
  valid_numbers = [float(n) for n in numbers]
  
  if not valid_numbers:
    return {
      "result": 0,
      "variance": 0
    }
    
  # Calculate mean
  mean = sum(valid_numbers) / len(valid_numbers)
  
  # Calculate variance
  variance = sum((x - mean) ** 2 for x in valid_numbers) / len(valid_numbers)
  
  return {
    "result": math.sqrt(variance),
    "variance": variance
  } 